src/pages/search.jsx

```diff
@@ -42,27 +43,38 @@ function shouldShowResults(conditions) {
     <NoResults page={ lastPage } searchTerm={ searchTerm } />;
 }
 
-function SearchPage(props) {
-  if (
-    typeof window !== 'undefined' &&
-    !('Promise' in window)
-  ) {
-    return <NoSupport />;
+class SearchPage extends PureComponent {
+  constructor() {
+    super();
+  }
+
+  componentWillUnmount() {
+    const { resetSearch } = this.props;
+    resetSearch('');
+  }
+
+  render() {
+    if (
+      typeof window !== 'undefined' &&
+      !('Promise' in window)
+    ) {
+      return <NoSupport />;
+    }
+    const { isSearching, lastPage, results, searchTerm } = this.props;
+    return (
+      <div>
+        <Helmet>
+          <title>Search | freeCodeCamp Guide</title>
+        </Helmet>
+        <h2 className='colourDarkGrey'>Search Results</h2>
+        {
+          (isSearching && !results.length) ?
+            <ResultsSkeleton /> :
+            shouldShowResults({ results, lastPage, searchTerm })
```

Comment:
Now this page is a `class`, we should make this function a method on the class. By doing so, we will not need to pass props to it manually, as they will be available via `this.props`

Reply:
Ok, I understand your comment. Just one question. What is the best practice? Moving the method "shouldShowResults" before "componentWillUnmount" or after.

Reply:
I think we should go with 
```jsx
class Search extends PureComponent {
  // constructor

  // component lifestyle hooks

  // component methods

  // render() {...}
}
```

Thanks for asking @marekdano

Reply:
@Bouncey Thanks for your answer. I've just pushed my code. Hopefully it's good ;)